Sami/Parser/NodeVisitor.php

```diff
@@ -151,8 +152,12 @@ protected function addMethod(ClassMethodNode $node)
             if ($param->default) {
                 $parameter->setDefault($this->context->getPrettyPrinter()->prettyPrintExpr($param->default));
             }
-            if ((string) $param->type) {
-                $parameter->setHint($this->resolveHint(array(array((string) $param->type, false))));
+            if ($type = (string) $param->type) {
+                if( $param->type instanceof FullyQualified && strpos($type, '\\') !== 0 ) {
```

Comment:
So the `$param->type` is sometimes a string with `string` or other scalar type and sometimes it's a `FullyQualified` class instance if this is class or interface (I wonder how PhpParser distinguishes between two class/interface or a scalar type).